packages/atlas/src/components/_crt/RevenueShareWidget/RevenueShareWidget.tsx

```diff
@@ -8,34 +10,32 @@ import { NumberFormat } from '@/components/NumberFormat'
 import { Text } from '@/components/Text'
 import { Button } from '@/components/_buttons/Button'
 import { InfoBox, Wrapper } from '@/components/_crt/RevenueShareWidget/RevenueShareWidget.styles'
+import { useBlockTimeEstimation } from '@/hooks/useBlockTimeEstimation'
 import { useJoystream } from '@/providers/joystream'
 import { useSnackbar } from '@/providers/snackbars'
 import { useTransaction } from '@/providers/transactions/transactions.hooks'
-import { useUser } from '@/providers/user/user.hooks'
 import { formatDateTime } from '@/utils/time'
 
 export type RevenueShareWidgetProps = {
   tokenId: string
   tokenName: string
-  userShare: number
-  userTokens: number
-  shareEndDate: Date
-  onClaimShare?: () => void
-  status: 'active' | 'upcoming' | 'locked' | 'unlocked'
+  revenueShare: GetTokenRevenueSharesQuery['revenueShares'][number]
+  memberId: string
 }
-export const RevenueShareWidget = ({
-  userShare,
-  userTokens,
-  tokenName,
-  onClaimShare,
-  shareEndDate,
-  status,
-  tokenId,
-}: RevenueShareWidgetProps) => {
+export const RevenueShareWidget = ({ tokenName, tokenId, revenueShare, memberId }: RevenueShareWidgetProps) => {
   const { joystream, proxyCallback } = useJoystream()
   const handleTransaction = useTransaction()
-  const { memberId } = useUser()
   const { displaySnackbar } = useSnackbar()
+  const { convertBlockToMsTimestamp, currentBlock } = useBlockTimeEstimation()
+  const memberStake = revenueShare.stakers.find((stakers) => stakers.account.member.id === memberId)
+  const status =
+    revenueShare.createdIn > currentBlock
```

Comment:
I think it was already created if it exists so
```suggestion
    revenueShare.startsAt > currentBlock
```